fmf/base.py

```diff
@@ -116,88 +157,92 @@ def commit(self):
         except IOError as error:
             raise utils.FormatError(
-                "Unable to detect format version: {0}".format(error))
+                f"Unable to detect format version: {error}")
         except ValueError:
             raise utils.FormatError("Invalid version format")
 
-    def _merge_plus(self, data, key, value, prepend=False):
+    def _merge_plus(self, data: TreeData, key: str,
+                    value: DataType, prepend: bool = False) -> None:
         """ Handle extending attributes using the '+' suffix """
-        # Nothing to do if key not in parent
-        if key not in data:
-            data[key] = value
-            return
-        # Use the special merge for merging dictionaries
-        if type(data[key]) == type(value) == dict:
-            self._merge_special(data[key], value)
-            return
-        # Attempt to apply the plus operator
         try:
+            # Nothing to do if key not in parent
+            if key not in data:
+                data[key] = value
+                return
+            # Use the special merge for merging dictionaries
+            data_val = data[key]
+            if type(data_val) == type(value) == dict:
+                self._merge_special(data_val, value)
+                data[key] = data_val
+                return
+            # Attempt to apply the plus operator
             if prepend:
-                data[key] = value + data[key]
+                data_val = value + data_val  # type: ignore
             else:
-                data[key] = data[key] + value
-        except TypeError as error:
-            raise utils.MergeError(
-                "MergeError: Key '{0}' in {1} ({2}).".format(
-                    key, self.name, str(error)))
+                data_val = data_val + value  # type: ignore
+            data[key] = data_val
+        except TypeError as err:
+            raise utils.MergeError(f"MergeError: Key '{key}' in {self.name}.") from err
 
-    def _merge_minus(self, data, key, value):
+    def _merge_minus(self, data: TreeData, key: str, value: DataType) -> None:
         """ Handle reducing attributes using the '-' suffix """
-        # Cannot reduce attribute if key is not present in parent
-        if key not in data:
-            data[key] = value
-            raise utils.MergeError(
-                "MergeError: Key '{0}' in {1} (not inherited).".format(
-                    key, self.name))
-        # Subtract numbers
-        if type(data[key]) == type(value) in [int, float]:
-            data[key] = data[key] - value
-        # Replace matching regular expression with empty string
-        elif type(data[key]) == type(value) == type(""):
-            data[key] = re.sub(value, '', data[key])
-        # Remove given values from the parent list
-        elif type(data[key]) == type(value) == list:
-            data[key] = [item for item in data[key] if item not in value]
```

Comment:
Yeah, this `type(...) == type(...)` seems to be common - it's definitely a wrong way, and since you're moving those lines around, and it is definitely related to types of values, I'd propose replacing these tests with proper `isinstance()` calls.